connectors/sources/sharepoint_online.py

```diff
@@ -82,6 +82,15 @@
 DELTA_NEXT_LINK_KEY = "@odata.nextLink"
 DELTA_LINK_KEY = "@odata.deltaLink"
 
+# See https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Model/SharePoint/Core/Public/Enums/PermissionKind.cs
+VIEW_ITEM_MASK = 0x1
+
+# See https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Model/SharePoint/Core/Public/Enums/PermissionKind.cs
+VIEW_PAGE_MASK = 0x20000
+
+# See https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Model/SharePoint/Core/Public/Enums/RoleType.cs
+VIEW_ROLE_TYPES = [2, 3, 4, 5, 6, 7, 0xFF]
```

Comment:
```suggestion
READER = 2
CONTRIBUTOR = 3
WEB_DESIGNER = 4
ADMINISTRATOR = 5
EDITOR = 6
REVIEWER = 7
SYSTEM = 0xFF
VIEW_ROLE_TYPES = [READER, CONTRIBUTOR, WEB_DESIGNER, ADMINISTRATOR, EDITOR, REVIEWER, SYSTEM]
```

Rationale: https://refactoring.guru/replace-magic-number-with-symbolic-constant
We could also add a comment, but nothing blocking ofc.